games/godot/Portfile

```diff
@@ -22,24 +22,52 @@ long_description    Godot Engine is a cross-platform game engine for \
                     (Linux, macOS, Windows), as well as mobile \
                     (Android, iOS) and web-based (HTML5) platforms.
 
-checksums           rmd160  4ff4f9c4798d67cf6cf4d63c38bed84d400daa8e \
+if {$subport eq ${name}} {
+    github.setup    godotengine ${name} 3.3 "" -stable
+    version         3.3.0
+
+    checksums       rmd160  4719f38a85cf339fa6c66d5849b972cd22c0a1ed \
+                    sha256  a65e296c900f442e19600853896f8f742a1414cc8864f9f6e45eeabc9c4b3362 \
+                    size    29432959
+
+    compiler.blacklist-append {clang < 802}
+}
+
+subport ${name}-3.2 {
```

Comment:
why not do the version selection based on `os.major` for the user in the Portfile directly instead of adding this subport and  the `known_fail` stuff and notes that tell the user the latest supported version for macOS 10.9.10.11?

Reply:
The upstream authors dropped support for macOSes version 10.11 and older starting in Godot 3.3.0. Recently I haven't had time to figure out how to get 3.3.x working on macOS 10.9–10.11, so I created the 3.2 subport in order to allow 10.9–10.11 to continue to have a working Godot in MacPorts. As far as why I don't do the version selection based on `os.major`: I feel like having Godot 3.2 available to users with macOS 10.12+ could be of benefit, in case users decide they need to continue using the older version of Godot. MacPorts continues to allow users with newer versions of macOS to install older versions of `perl`, `clang`, etc., right?

The `known_fail` is for macOS 10.8 and older. I've been instructed by many of the senior MacPorts devs to add `known_fail` to my portfiles in order to lighten the load on the builders.